eng/common/pipelines/templates/steps/check-spelling.yml

```diff
@@ -0,0 +1,38 @@
+# Checks spelling of files that changed between the current state of the repo
+# and some ref (branch, tag, etc.) or commit hash. Only runs on PRs.
+# ContinueOnError - true: Pipeline warns on spelling error 
+#                   false: Pipeline fails on spelling error
+# TargetRef       - Target ref (e.g. master) to compare to create file change
+#                   list
+# CspellConfigPath - Path to cspell.json config location
+
+parameters:
+  ContinueOnError: true
+  TargetRef: $(System.PullRequest.TargetBranch)
+  CspellConfigPath: ./.vscode/cspell.json
+
+steps:
+  - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+    # Devops uses a git config that changes ref names. 
```

Comment:
Is there any reason the approach that we are using in https://github.com/Azure/azure-sdk-tools/pull/1415 (and https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/scripts/get-markdown-files-from-changed-files.ps1#L5) will not work here? I'd like to use similar strategy across use cases so we aren't doing this differently. It is starting to seem like we need a common way to get a list of the changed files.

Reply:
Indeed it does work: https://dev.azure.com/azure-sdk/public/_build/results?buildId=741805&view=logs&j=a129effc-2dd1-54d1-fb5a-ad7bdc0e851d&t=617ceecd-bf48-5ce3-83e2-7468b1632a13&l=13 

My concern here is that this approach couples the pipeline logic to how DevOps configures the repo today. My proposal is not as coupled. In the interest of consistency and simplicity I'll use the approach we're already using. If we want to use what I proposed here we can undertake that in a separate effort.

Reply:
I kind of agree with you about the coupling to Devops, in particular the particular variables that Devops sets. We should be able to decouple that in the future by coupling that target branch outside of the script and then passing in the new value. It might be worth filing a follow-up issue to try and refactor that logic out and share it across our different usages.

Reply:
https://github.com/Azure/azure-sdk-tools/issues/1437